vars/createGithubIssue.groovy

```diff
@@ -6,77 +6,34 @@
-        def currentVersion = yamlFile.build.version
-
-        for(component in yamlFile.components){
-            if (failedComponents.contains(component.name)) {
-                println("Component ${component.name} failed, creating github issue")
-                compIndex = failedComponents.indexOf(component.name)
-                create_issue(component.name, component.repository, currentVersion, failureMessages[compIndex])
-                sleep(time:3,unit:"SECONDS")
-            }
-        }
-    }
-}
-
-def create_issue(component, url, currentVersion, failedMessage){
-    def versionLabel = "v${currentVersion}"
-    def label = "autocut"
-
-    def message = """***Received Error***: **${failedMessage}**.
-                      The distribution build for ${component} has failed.
-                      Please see build log at ${BUILD_URL}consoleFull""".stripIndent()
-
 
+/** Library to create GitHub issue across opensearch-project repositories.
+ @param Map args = [:] args A map of the following parameters
+ @param args.githubUrl <required> - GitHub repository URL to create issue
+ @param args.issueTitle <required> - GitHub issue title
+ @param args.issueBody <required> - GitHub issue body
+ @param args.label <optional> - GitHub issue label to be attached along with 'untriaged'. Defaults to autocut.
+ */
+void call(Map args = [:]) {
+    label = args.label ?: 'autocut'
     try {
         withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
-            def issues = sh (
-                    script: "gh issue list --repo ${url} -S \"[AUTOCUT] OS Distribution Build Failed for ${component}-${currentVersion} in:title\" --label ${label}",
+            def issues = sh(
+                    script: "gh issue list --repo ${args.githubUrl} -S \"${args.issueTitle} in:title\" --label ${label}",
                     returnStdout: true
             )
 
-            def hasLabel = sh (
-                    script: "gh label list --repo ${url} -S ${versionLabel}",
-                    returnStdout: true
-            )
-
-            if (hasLabel){
-                label = "\"autocut,${versionLabel}\""
+            if (issues) {
+                println('Issue already exists in the repository, skipping.')
             }
-
-            if (issues){
-                println("Issue already exists in the repository, skipping.")
-            } else {
-                sh (
-                        script: "gh issue create --title \"[AUTOCUT] OS Distribution Build Failed for ${component}-${currentVersion}\" --body \"${message}\" --label ${label} --label \"untriaged\" --repo ${url}",
-                        returnStdout: true
+            else {
+                sh(
+                    script: "gh issue create --title \"${args.issueTitle}\" --body \"${args.issueBody}\" --label ${label} --label \"untriaged\" --repo ${args.githubUrl}",
```

Comment:
I believe all the repo have the workflow to attach `untriaged` label on new issues, do we need to add this label here?
Also, we need to attach the version label as well.

Reply:
Yeah version label is added in https://github.com/opensearch-project/opensearch-build-libraries/pull/188/files#diff-aecaaf0a03223127fa0678afd20a2ab971d2508c86ff41197a3be915931b00c4R51
Not forcing it in base library as it can be used later for any issue creation.

Reply:
Regarding untriaged, keeping it for now no harm to attach that label when an issue is created